python/mlcroissant/mlcroissant/_src/operation_graph/operations/download_test.py

```diff
@@ -68,3 +75,38 @@ def test_insert_credentials():
             username=None,
             password="my/password",
         )
+
+def test_get_hash_obj_md5():
+    file_object = FileObject(
+            md5="12345",
+        )
+    hash_obj = _get_hash_obj(file_object)
+
+    assert isinstance(hash_obj, type(hashlib.md5()))
+
+def test_get_hash_obj_sha256():
+    file_object = FileObject(
+            sha256="12345",
+        )
+    hash_obj = _get_hash_obj(file_object)
+
+    assert isinstance(hash_obj, type(hashlib.sha256()))
+
+def test_checking_hash_correctly():
```

Comment:
I see two possible issues with this unit test:

- The test is non-hermetic (meaning it downloads data from the internet).
- The test checks the function by using the content of the function itself (you copy/pasted the logic from the code to the test).

Typically we would use mocking instead ([see the documentation](https://docs.python.org/3/library/unittest.mock.html)).

Warning: this is "pseudo-code" which I haven't tested.

```python
@mock.patch.object(requests, 'get')
def test_checking_hash_correctly(get_hash_mock):
  # Assert that the file doesn't exist
  # assert ...
  node = FileObject(
    sha256='c617db2c7470716250f6f001be51304c76bcc8815527ab8bae734bdca0735737',
  )
  Download(node=node)()
  # Assert that the file has been written in the good path with the proper hash 
  # assert ...
```

Also, other tests in the codebase should fail. So I recommend you focus on those tests. We can write a unit test later in another PR if needed.